Dockerfile

```diff
@@ -249,26 +249,28 @@ RUN set -x \
 	&& go build -v -o /usr/local/bin/rsrc github.com/akavel/rsrc \
 	&& rm -rf "$GOPATH"
 
+# Upload docker source
+COPY . /go/src/github.com/docker/docker
+
```

Comment:
Two reasons:
1. We clone only once, and since we need to build the files many times for packaging reason (unless you know of a way to prevent having to build for every package) we save ourselves the clone everytime.
2. This limits where we store the commit hash for these two packages, if we don't do it this way we need to clone many times and we would also need the commit hash in multiple places, which could lead to the situation where we get out of sync.

for number two, we could probably get around that by doing some awk magic the way we deal with GO_VERSION in the packaging generate.sh scripts.

We can reclone the same code everytime we build the sub packages, it will be slower, but we build packages a lot less then we build the master dockerfile when developing.

Ideally I would like to compile once and save the binaries somewhere and use them in the packaging, but I don't think that is possible?  If someone knows better please let me know. 

Maybe we can run 'cross' before we run bundle and copy them out of the bundles/cross/ arch directories ? Not sure if that would give us the correct binaries or not.

Once again, I'm fine with changing this, just need to know the best way, and don't want to make the change a bunch of times, so let's discuss here first, then once everyone agrees, I'll switch.